tests/Http/AcceptHeaderTest.php

```diff
@@ -0,0 +1,163 @@
+        $this->assertEquals(1, $compareValue->invokeArgs(null, ['type/subtype', 'type/*']));
+        $this->assertEquals(-1, $compareValue->invokeArgs(null, ['type/*', 'type/subtype']));
+    }
+
+    public function testHeaderItem()
+    {
+        $item = new AcceptHeaderItem('text/html', 0.9, ['t' => 'test']);
+
+        $this->assertEquals('text/html; q=0.9; t=test', $item);
+        $this->assertEquals(0.9, $item->getQuality());
+        $this->assertEquals(['t' => 'test'], $item->getParams());
+
+        $item = $item->withValue('application/json');
+
+        $this->assertEquals('application/json', $item->getValue());
+
+        $item = $item->withQuality(0);
+
+        $this->assertEquals(0, $item->getQuality());
+
+        $item = $item->withParams(['n' => 'new']);
+
+        $this->assertEquals(['n' => 'new'], $item->getParams());
+
+        $wrongParamsItem = AcceptHeaderItem::fromString('text/html; q');
+
+        $this->assertEquals([], $wrongParamsItem->getParams());
+        $this->assertEquals(1.0, $wrongParamsItem->getQuality());
+
+        AcceptHeaderItem::fromString('');
+    }
+
+    public function testHeaderConstructing()
+    {
+        $acceptCharset = new AcceptHeader(['*', 'UTF-8']);
+
+        $this->assertFalse($acceptCharset->has('unicode'));
+        $this->assertNull($acceptCharset->get('unicode'));
+
+        $sorted = $acceptCharset->sorted();
+        $this->assertEquals('UTF-8', $sorted[0]);
+        $this->assertEquals('*', $sorted[1]);
+        $this->assertEquals('UTF-8, *', $acceptCharset);
+
+        $acceptCharset = $acceptCharset->add(AcceptHeaderItem::fromString('unicode'));
+
+        $this->assertTrue($acceptCharset->has('unicode'));
+        $this->assertNotNull($acceptCharset->get('unicode'));
+
+        $sorted = $acceptCharset->sorted();
+        $this->assertEquals('UTF-8', $sorted[0]);
+        $this->assertEquals('unicode', $sorted[1]);
+        $this->assertEquals('*', $sorted[2]);
+        $this->assertEquals('UTF-8, unicode, *', $acceptCharset);
+
+        $acceptHeader = AcceptHeader::fromString('text/html; q=0.8, */*; q=0.7');
+        $acceptHeader = $acceptHeader->add('application/json');
+
+        $sorted = $acceptHeader->sorted();
+        $this->assertEquals('application/json', $sorted[0]);
+        $this->assertEquals('text/html; q=0.8', $sorted[1]);
+        $this->assertEquals('*/*; q=0.7', $sorted[2]);
+    }
+
+    public function testHeader1()
+    {
+        $acceptHeader = AcceptHeader::fromString('audio/*; q=0.2, audio/basic')->sorted();
+        $this->assertEquals('audio/basic', (string) $acceptHeader[0]);
+        $this->assertEquals('audio/*; q=0.2', (string) $acceptHeader[1]);
+    }
+
+    public function testHeader2()
+    {
+        $acceptHeader = AcceptHeader::fromString(
+            'text/*;q=0.3, text/html;q=0.7, text/html;level=1, text/html;level=2;q=0.4, */*;q=0.5'
+        )->sorted();
+
+        $this->assertEquals('text/html; level=1', $acceptHeader[0]);
```

Comment:
I propose also to have `assertCount` assertions while having one-by-one assertions for each **expected** array item - you may receive more elements and your test will not show them